actors/user/src/main/java/org/sunbird/user/actors/TenantMigrationActor.java

```diff
@@ -401,4 +409,73 @@ private void rejectMigration(Request request) {
     response.put(JsonKey.SUCCESS, true);
     sender().tell(response, self());
   }
+
+
+  private void selfMigrate(Request request) {
+    String userId = (String) request.getRequest().get(JsonKey.USER_ID);
+    String extUserId = (String) request.getRequest().get(JsonKey.USER_EXT_ID);
+    ShadowUser shadowUser = MigrationUtils.getRecordByUserId(userId);
+    if(null==shadowUser){
+      ProjectCommonException.throwClientErrorException(ResponseCode.invalidUserId);
+    }
+    if(shadowUser.getClaimStatus() == ClaimStatus.CLAIMED.getValue()){
+      ProjectCommonException.throwClientErrorException(ResponseCode.unAuthorized);
+    }
+     if (StringUtils.equalsIgnoreCase(shadowUser.getUserExtId(), extUserId)) {
+      prepareMigrationRequest(request,shadowUser,userId,extUserId);
+      ProjectLogger.log("TenantMigrationActor:selfMigrate:request prepared for user migration:"+request.getRequest(),LoggerEnum.INFO.name());
+      migrateUser(request);
+      Map<String, Object> propertiesMap = new HashMap<>();
```

Comment:
instead of hashMap , if scope is within block we can use weakHashMap

Reply:
done